src/app/modules/shared_models/currency_amount.nim

```diff
@@ -0,0 +1,63 @@
+import NimQml, strformat
+
+QtObject:
+  type CurrencyAmount* = ref object of QObject
+    amount: float64
+    symbol: string
+    displayDecimals: int
+    stripTrailingZeroes: bool
+
+  proc setup(self: CurrencyAmount) =
+    self.QObject.setup
+
+  proc delete*(self: CurrencyAmount) =
+    self.QObject.delete
+
+  proc newCurrencyAmount*(
+    amount: float64,
+    symbol: string,
+    displayDecimals: int,
+    stripTrailingZeroes: bool,
+  ): CurrencyAmount =
+    new(result, delete)
+    result.setup
+    result.amount = amount
+    result.symbol = symbol
+    result.displayDecimals = displayDecimals
+    result.stripTrailingZeroes = stripTrailingZeroes
+
+  proc `$`*(self: CurrencyAmount): string =
+    result = fmt"""CurrencyAmount(
+      amount: {self.amount},
+      symbol: {self.symbol},
+      displayDecimals: {self.displayDecimals},
+      stripTrailingZeroes: {self.stripTrailingZeroes}
+      ]"""
+
+  proc getAmount*(self: CurrencyAmount): float64 =
+    return self.amount
+
+  # QtProperty[float] is broken, data received in QML is garbage
```

Comment:
Calling `setup` last in the constructor should fix this issue; see the above comment. If it doesn't work, let's debug this together.

Reply:
I see now @caybro's finding. I guess I have to understand better how the QML-Nim library work.